2023/src/day5/day5main.rs

```diff
@@ -1,24 +1,41 @@
 
-use crate::{util::input_util::load_day_input, day5::almanac_parse::parse_almanac};
+use crate::{
+  util::input_util::load_day_input,
+  day5::almanac_parse::parse_almanac,
+};
 
-const DAY_5_INPUT_FILE_NAME: &str = "day5_test.txt";
+const DAY_5_INPUT_FILE_NAME: &str = "day5.txt";
 
 pub fn day5_main() {
-  println!("~ Day 5 ~");
+  println!("~ Day 5 ~\n");
 
   let input_lines: Vec<String> = load_day_input(DAY_5_INPUT_FILE_NAME)
     .into_iter()
     .filter(|line| line.len() > 0)
     .collect();
-  let amc = parse_almanac(&input_lines);
-  println!("Seeds:\n{}", list_to_str(&amc.seeds));
-  for amc_entry in amc.amc_entries.iter() {
-    println!();
-    println!("{:#?}", amc_entry.kind);
-    for amc_map in amc_entry.amc_maps.iter() {
-      println!("{} {} {}", amc_map.dest_start, amc_map.src_start, amc_map.range_len);
+  let amc = parse_almanac(input_lines);
+
+  let mut locations: Vec<u64> = Vec::new();
```

Comment:
You can probably omit the `: Vec<u64>` here and the compiler should be able to figure it out.  It's almost kind of a fun little game seeing how many annotations you can get away with omitting.

Reply:
oh gosh you were right lol. The compiler is super useful. I wish I knew all of the scenarios where type inference is possible ... but I guess that's what the compiler is for. I'll intuit it eventually I guess!

Reply:
Generally it can figure things out based on:
* usage later in the block (e.g. pushing only `u64`s into a `Vec`)
* the value being used as the return value, since the return type is specified in the function signature
* similarly, if you specify something like `.collect::<Vec<u64>>()` at the end of an iterator chain, you might be able to omit stuff when working in a block within a `map` call

There's definitely other instances, it's surprisingly robust.